spec/lib/efile/nj/nj_flat_eitc_eligibility_spec.rb

```diff
@@ -0,0 +1,333 @@
+
+    date_older_than_65 = Date.new(year_65+1, 1, 1) # 1/1/1960
+    date_younger_than_65 = Date.new(year_65+1, 1, 2) # 1/2/1960
+
+    date_older_than_25 = Date.new(year_25+1, 1, 1) # 1/1/2000
+    date_younger_than_25 = Date.new(year_25+1, 1, 2) # 1/2/2000
+
+    date_older_than_18 = Date.new(year_18, 12, 31) # 12/31/2006
+    date_younger_than_18 = Date.new(year_18+1, 1, 1) # 1/1/2007
+
+    context "when mfj" do
+      [
+        # 2024: Taxpayer is born on or before 1/1/1960. If MFJ, both taxpayers have to meet the requirement.
+        { primary_birth_date: date_older_than_65, spouse_birth_date: date_older_than_65, expected: true },
+        { primary_birth_date: date_younger_than_65, spouse_birth_date: date_older_than_65, expected: false },
+        { primary_birth_date: date_older_than_65, spouse_birth_date: date_younger_than_65, expected: false },
+        { primary_birth_date: date_younger_than_65, spouse_birth_date: date_younger_than_65, expected: false },
+
+        # 2024: Taxpayer is born on or after 1/1/2000. If MFJ, both taxpayers have to meet the requirement.
+        { primary_birth_date: date_younger_than_25, spouse_birth_date: date_younger_than_25, expected: true },
+        { primary_birth_date: date_younger_than_25, spouse_birth_date: date_older_than_25, expected: false },
+        { primary_birth_date: date_older_than_25, spouse_birth_date: date_younger_than_25, expected: false },
+        { primary_birth_date: date_older_than_25, spouse_birth_date: date_older_than_25, expected: false },
+
+        # 2024: taxpayer is born on or before 12/31/2006. If MFJ, only one taxpayer has to meet the requirement
+        { primary_birth_date: date_younger_than_18, spouse_birth_date: date_younger_than_18, expected: false },
+        { primary_birth_date: date_older_than_18, spouse_birth_date: date_younger_than_18, expected: true },
+        { primary_birth_date: date_younger_than_18, spouse_birth_date: date_older_than_18, expected: true },
+        { primary_birth_date: date_older_than_18, spouse_birth_date: date_older_than_18, expected: true },
+      ].each do |test_case|
+        context "when #{test_case}" do
+          let(:intake) { create(:state_file_nj_intake, :married_filing_jointly) }
+          it "returns #{test_case[:expected]}" do
+            allow(intake).to receive(:primary_birth_date).and_return(test_case[:primary_birth_date])
+            allow(intake).to receive(:spouse_birth_date).and_return(test_case[:spouse_birth_date])
+
+            expect(Efile::Nj::NjFlatEitcEligibility.meets_age_requirements?(intake)).to eq(test_case[:expected])
+          end
+        end
+      end
+    end
+
+    context "when not mfj" do
+      [
+         # 2024: Taxpayer is born on or before 1/1/1960
+        { primary_birth_date: date_older_than_65, expected: true },
+        { primary_birth_date: date_younger_than_65, expected: false },
+
+        # 2024: Taxpayer is born on or after 1/1/2000
```

Comment:
Same comment here - I suspect comment is stale

Reply:
fixed